### DISCUSSION TITLE
Setting Up Lighthouse CI in Bamboo

### DISCUSSION THREAD

**Discussion by jowherr** [GoogleChrome/lighthouse-ci#661] [Q&A]
Hi,

I'm fairly new to Bamboo and really keen on implementing Lighthouse CI in our Pipeline. Unfortunately, even if I create an NPM tasks to install lighthouse ci, whenever I run lhci autorun, it's alway producing an error 

/home/bamboo/bamboo-agent-home/temp/PT-LIG0-JOB1-9-ScriptBuildTask-7505254340593389088.sh: 1: /home/bamboo/bamboo-agent-home/temp/PT-LIG0-JOB1-9-ScriptBuildTask-7505254340593389088.sh: **lhci: not found**

Would appreciate any assistance on this issue.

**Comment by jowherr** [comment]
[plan-249430172-JOB1-25.log](https://github.com/GoogleChrome/lighthouse-ci/files/6928839/plan-249430172-JOB1-25.log)

Some progress. I'm getting this exception when calling lhci autorun:



04-Aug-2021 05:36:11 | /opt/node-v10.15.3-linux-x64/bin/lhci -> /opt/node-v10.15.3-linux-x64/lib/node_modules/@lhci/cli/src/cli.js
-- | --
04-Aug-2021 05:36:11 | + @lhci/cli@0.8.0
04-Aug-2021 05:36:11 | added 333 packages from 205 contributors in 6.86s
04-Aug-2021 05:36:12 | /opt/node-v10.15.3-linux-x64/lib/node_modules/@lhci/cli/src/cli.js:55
04-Aug-2021 05:36:12 | ...argv,
04-Aug-2021 05:36:12 | ^^^
04-Aug-2021 05:36:12 | SyntaxError: Unexpected token ...
04-Aug-2021 05:36:12 | at Object.exports.runInThisContext (vm.js:73:16)
04-Aug-2021 05:36:12 | at Module._compile (module.js:543:28)
04-Aug-2021 05:36:12 | at Object.Module._extensions..js (module.js:580:10)
04-Aug-2021 05:36:12 | at Module.load (module.js:488:32)
04-Aug-2021 05:36:12 | at tryModuleLoad (module.js:447:12)
04-Aug-2021 05:36:12 | at Function.Module._load (module.js:439:3)
04-Aug-2021 05:36:12 | at Module.runMain (module.js:605:10)
04-Aug-2021 05:36:12 | at run (bootstrap_node.js:422:7)
04-Aug-2021 05:36:12 | at startup (bootstrap_node.js:143:9)
04-Aug-2021 05:36:12 | at bootstrap_node.js:537:3

**Reply by patrickhulce** [reply]
What versions of node do you have installed on the machine? I see `node-v10.15.3` (which by the way is too low, node 12 is required for lighthouse), but the error you received indicates the use of a *very* old node version to actually execute.

**Reply by jowherr** [reply]
Thank you for responding, Patrick. The available versions right now in the Agents are Node 10 and Node 6, I've requested for version 12 (or above) to be installed.

**Reply by patrickhulce** [reply]
Ah that would do it. This error indicates that node 6 was used to execute node LHCI. Once you have node 12, make sure you're using *that* binary to execute lighthouse.
